### PR TITLE
fix: add minimum TF version requirements

### PR DESCRIPTION
**Pull request by apeabody**
- add PoC lint terraform minimum validate
- add tf needed `required_version`s

### CODE REVIEW

**Comment by apeabody** [comment]
```
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; 
        	            	Error: Error waiting to create Feature: Error waiting for Creating Feature: Error code 13, message: an internal error has occurred
        	            	
        	            	  with module.env.google_gke_hub_feature.mcs,
        	            	  on ../../modules/env_baseline/mcg.tf line 34, in resource "google_gke_hub_feature" "mcs":
        	            	  34: resource "google_gke_hub_feature" "mcs" {
        	            	}
        	Test:       	TestFleetscope/development
```

**Comment by apeabody** [comment]
```
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; 
        	            	Error: googleapi: Error 403: Google Compute Engine: Required 'compute.networks.get' permission for 'projects/eab-vpc-production-803a/global/networks/eab-vpc-production'.
        	            	Details:
        	            	[
        	            	  {
        	            	    "@type": "type.googleapis.com/google.rpc.RequestInfo",
        	            	    "requestId": "0x187f58efa15a8957"
        	            	  }
        	            	]
        	            	, forbidden
        	            	
        	            	  with module.env.module.gke["eab-production-region01"].google_container_cluster.primary,
        	            	  on .terraform/modules/env.gke/modules/beta-private-cluster/cluster.tf line 22, in resource "google_container_cluster" "primary":
        	            	  22: resource "google_container_cluster" "primary" {
        	            	}
        	Test:       	TestMultitenant/production
```

**Comment by apeabody** [comment]
```
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; 
        	            	Error: Error waiting for creating GKE cluster: Failed to create cluster
        	            	
        	            	  with module.env.module.gke["eab-production-region01"].google_container_cluster.primary,
        	            	  on .terraform/modules/env.gke/modules/beta-private-cluster/cluster.tf line 22, in resource "google_container_cluster" "primary":
        	            	  22: resource "google_container_cluster" "primary" {
        	            	}
        	Test:       	TestMultitenant/production
```

**Comment by apeabody** [comment]
`" - statefulset/ledger-db failed. Error: container postgres terminated with exit code 137."`

**Comment by apeabody** [comment]
```
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; 
        	            	Error: Error waiting to create Feature: Error waiting for Creating Feature: Error code 13, message: an internal error has occurred
        	            	
        	            	  with module.env.google_gke_hub_feature.acm_feature,
        	            	  on ../../modules/env_baseline/acm.tf line 44, in resource "google_gke_hub_feature" "acm_feature":
        	            	  44: resource "google_gke_hub_feature" "acm_feature" {
        	            	}
        	Test:       	TestFleetscope/production
```

**Comment by apeabody** [comment]
`FAIL: TestSourceCymbalBank/../../../6-appsource/cymbal-bank/ledger-ledgerwriter`

**Comment by apeabody** [comment]
`FAIL: TestSourceCymbalBank/../../../6-appsource/cymbal-bank/ledger-transactionhistory`

File: 4-fleetscope/envs/development/versions.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  required_version = ">= 1.6"
```
Comment (yliaog) [inline 4-fleetscope/envs/development/versions.tf:18]: it is not 1.3 ? what makes 1.6 a requirement?

Comment (apeabody) [reply 4-fleetscope/envs/development/versions.tf:18]: Yeah @yliaog.  For future discussion: `import` blocks were added in v1.5, and the ability to use variables in `import` block ids in v1.6

https://github.com/GoogleCloudPlatform/terraform-google-enterprise-application/blob/main/4-fleetscope/envs/development/main.tf#L21

Comment (yliaog) [reply 4-fleetscope/envs/development/versions.tf:18]: Got it.

**Review by yliaog — APPROVED** [review]

**Comment by apeabody** [comment]
`TestSourceCymbalBank/../../../6-appsource/cymbal-bank/accounts-contacts`

**Pull request merged**